lib/matplotlib/cm.py

```diff
@@ -193,6 +193,49 @@ def unregister(self, name):
                              "colormap.")
         self._cmaps.pop(name, None)
 
+    def ensure_colormap(self, cmap, *, default=None):
```

Comment:
I'm surprised this isn't just `colormaps.get_cmap`?

Reply:
That is what we deprecated!  I wanted to add the ability to control the default and to raise `KeyError` rather than `ValueError`.

Maybe I am over thinking.

Reply:
I think we deprecated `cm.get_cmap`...

Reply:
I'm confused.

Reply:
We used to have `matplotlib.cm.get_cmap`.  Its now deprecated.  

You are adding `matplotlib.cm.ColormapRegistry.ensure_colormap` (which is an instance of `matplotlib.cm.ColormapRegistry`, but 🤷 ).   I'm not understanding why we aren't calling this new method on `matplotlib.cm.ColormapRegistry.get_cmap`, so users will end up spelling our global registry as `matplotlib.colormaps.get_cmap()`

But maybe I am confused....

Reply:
ah, sorry, I thought you were saying there was an exiting method we should have just used.

I went with the slightly longer name as I think we want to push most users to go for the `pml.colormaps[...]` path rather than going through this method.

Reply:
Yes, sorry, I was just questioning the name change.   But if you want a more esoteric name to discourage access, seems fair

Reply:
My 2 cents:

- `def ensure_colormap(self, cmap)` is enough. We can hard-code `None` -> `rcParams['image.cmap']` and do not need a `default` kwarg. This can easily be done on the user side via `ensure_colormap(mydefault if cmap is None)`.
- I'm inclined to stick with `get_cmap` naming for two reasons
  - The transition is more smooth: Use `matplotlib.colormaps.get_cmap()` instead of `matplotlib.cm.get_cmap()` looks like only moving the same functionality to another place (which is not 100% true because we drop the bogous `lut` parameter, but that's only used by a tiny fraction of users and I'm willing to accept this to have naming continuity for all normal users).
  - For backward compatibility we will still keep `pyplot.get_cmap()` around. With the proposed API and naming. `pyplot.get_cmap()` can be considered as a shortcut for `matplotlib.colormaps.get_cmap()`. This type of forwarding is pretty common in `pyplot` and conceptually simpler than having another function for retrieving a colormap.

    To make this analogy complete (and actually use in the implementation later) we still have to deprecate`plt.get_cmap(name, lut)` in favor of `plt.get_cmap(name).resampled(lut)`. But that change is independent of implementing this PR.

Reply:
I am convinced, changes pushed.